app/test/e2e/webshot-screenshot-post.spec.js

```diff
@@ -0,0 +1,81 @@
+const chai = require('chai');
+const nock = require('nock');
+const sinon = require('sinon');
+
+const { getTestServer } = require('./utils/test-server');
+const { stubPuppeteer, stubKoaSend } = require('./utils/stubs');
+
+chai.should();
+
+let requester;
+let sinonSandbox;
+
+nock.disableNetConnect();
+nock.enableNetConnect(process.env.HOST_IP);
+
+describe('GET /', () => {
+    before(async () => {
+        if (process.env.NODE_ENV !== 'test') {
+            throw Error(`Running the test suite with NODE_ENV ${process.env.NODE_ENV} may result in permanent data loss. Please use NODE_ENV=test.`);
+        }
+    });
+
+    beforeEach(() => { sinonSandbox = sinon.createSandbox(); });
+
+    describe('happy case', () => {
+        it('Takes a screenshot of the widget returning 200 OK with the URL for the pdf', async () => {
+            stubPuppeteer(sinonSandbox);
+            stubKoaSend(sinonSandbox, true);
+
+            requester = await getTestServer();
+
+            const response = await requester.get(`/api/v1/webshot?url=http://www.example.com&filename=newname`).send();
+            response.status.should.equal(200);
+            response.type.should.equal('application/pdf');
+        });
+
+        it('Takes a screenshot of the widget returning 200 OK with the URL for the png', async () => {
+            stubPuppeteer(sinonSandbox);
+            stubKoaSend(sinonSandbox, true);
+
+            requester = await getTestServer();
+
+            const response = await requester.get(`/api/v1/webshot?url=http://www.example.com&filename=newname&format=png`).send();
+            response.status.should.equal(200);
+            response.type.should.equal('application/png');
+        });
+    });
+
+    it('If puppeteer fails taking the screenshot returns 500 Internal Server Error', async () => {
```

Comment:
No know error case should return an "Internal server error" - if you-in-6-months get a bug report from a user saying "all I see is a message saying Internal Server Error", you'll have a poor starting point to find and fix the error.

Also, why 500? is it REALLY a server error? for example, if the URL provided by the user is not accessible, it should not be a 5xx error. please clarify

Reply:
and it goes without saying: if you need help, ping me on slack